generators/profilegen/src/profile_generator.rb

```diff
@@ -185,14 +214,16 @@ def get_client_file
     File.new(file_name, 'w')
   end
 
-  def check_for_availability(array_of_hash, key)
+  def check_and_delete_if_available(array_of_hash, key)
     array_of_hash.each do |hash_entry|
       if(hash_entry.keys.include?(:model_name))
         if(hash_entry[:model_name] == key)
+          array_of_hash.delete(hash_entry)
```

Comment:
how do you know you're keeping the latest in the hash? (I believe you mentioned you would keep latest only)